### PR TITLE
Add capabilities to dynamically  Bring Your Own Elastic IP for Nat Gateway

### PR DESCRIPTION
**Pull request by wvxavier**
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to <repo_name>?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the CONTRIBUTING.md doc.
- [x] I have added necessary documentation (if appropriate).
- [x] Any dependent changes have been merged and published in downstream modules.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by adenot** [comment]
Would be required to pass 3 allocation_ids when using multi-nat, right?

**Comment by lzrocha** [comment]
> Would be required to pass 3 allocation_ids when using multi-nat, right?

My questions are related to this as well

**Comment by adenot** [comment]
I'm ok if we import the EIPs into the module when we need instead of passing as a parameter, I think this is adding unecessary complexity, or is there another reason?

**Comment by wvxavier** [comment]
That's true.

I've converted the variable eip_allocation_id to a list so one can add multiple allocation ids when multi_nat is used.

**Comment by adenot** [comment]
Looks good. Can you just rename to `eip_allocation_ids`?

**Comment by wvxavier** [comment]
Sure thing, it's been renamed.
Also, I've tested the module using workspaces with and without EIP, looking good. ✅

**Review by lzrocha — DISMISSED** [review]

File: _variables.tf [hunk 1]
```diff
@@ -126,6 +126,17 @@ variable "kubernetes_clusters_type" {
   description = "Use either 'owned' or 'shared' for kubernetes cluster tags"
 }
 
+variable "byoip" {
+  type        = bool
+  default     = false
+  description = "Enable EC2 Bring your own IP (BYOIP) pool"
```
[the remark this answers is not in the document]
Comment (wvxavier) [reply _variables.tf:132]: Changing the description to: `Enable module to use your own Elastic IPs (Bring Your Own IP) `

**Review by lzrocha — APPROVED** [review]
Great Job @wvxavier. Thanks!

**Pull request merged**